lib/streamlit/DeltaGenerator.py

```diff
@@ -1876,6 +1881,39 @@ def slider(
             current_value = current_value[0] if single_value else current_value
         return current_value if single_value else tuple(current_value)
 
+    @_with_element
+    def file_uploader(self, element, label, type=['*'], key=None):
+
+        """Display a file uploader widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this file uploader is for.
+        type : string[] or None
+            Array of allowed extensions. (['.png', '.jpg'])
+            Default value: ['*']
+
+        Returns
+        -------
+        byte[] or None
+            The byte array of uploaded file or None if no one file is loaded.
+
+        Examples
+        --------
+        >>> file = st.file_uploader("Upload a image", type=([".png"]))
+        >>> if file != None:
+        >>>     st.image(file)
+
+        """
+        element.file_uploader.label = label
+        element.file_uploader.type[:] = type
+        element.file_uploader.max_upload_size = config.get_option("server.maxUploadSize")
+        _set_widget_id("file_uploader", element, user_key=key)
+        progress, data = self._file_manager.get_data(element.file_uploader.id)
+        element.file_uploader.progress = progress
+        return data if data != None else NoValue
```

Comment:
Also, this `NoValue` thing doesn't seem to be working for me.

When I run this:

```py
import streamlit as st

the_file = st.file_uploader("Upload a file")
'Result:', the_file
```

I get:

![image](https://user-images.githubusercontent.com/690814/68443651-75fec680-0189-11ea-9294-1565e67ab45a.png)

But I expected to get `Result: None`

Reply:
This is weird. It's working for me:
<img width="718" alt="Screen Shot 2019-11-08 at 12 46 29 PM" src="https://user-images.githubusercontent.com/22157341/68511759-664cb400-0255-11ea-9686-3672e79bcef9.png">